src/savegame_json.cpp

```diff
@@ -3808,6 +3821,18 @@ void submap::load( JsonIn &jsin, const std::string &member_name, bool rubpow_upd
             }
             partial_constructions[pt] = pc;
         }
+    } else if( member_name == "crafting_bills" ) {
+        jsin.start_array();
+        while( !jsin.end_array() ) {
+            crafting_bill cb;
+            tripoint pt;
+            jsin.read( pt );
+            jsin.start_array();
+            while( !jsin.end_array() ) {
+                cb.bills.emplace( recipe_id( jsin.get_string() ), jsin.get_int() );
```

Comment:
Probably problematic: the order of function argument evaluation is unspecified in c++, so this could read the int first and then the string. Need to first read them into temporaries to ensure correct reading order.